frame/election-provider-multi-phase/src/benchmarking.rs

```diff
@@ -208,7 +208,8 @@ frame_benchmarking::benchmarks! {
 		// assume a queued solution is stored, regardless of where it comes from.
 		<QueuedSolution<T>>::put(ready_solution);
 	}: {
-		let _ = <MultiPhase<T> as ElectionProvider<T::AccountId, T::BlockNumber>>::elect();
+		let res = <MultiPhase<T> as ElectionProvider<T::AccountId, T::BlockNumber>>::elect();
+		debug_assert!(res.is_ok());
```

Comment:
Inside benchmarking code we should use a normal assert. It is not part of on-chain execution.

Reply:
I asked @shawntabrizi about asserts in benchmarks https://github.com/paritytech/substrate/issues/8106#issuecomment-808080723

And as I understood him I should use debug_asserts?

Reply:
His argumentation is that the code is called by the tests anyways and should therefore be run as debug build. However, I would argue that we want the asserts to be run during the actual benchmarking which runs with a different parameterization. Also, I am not sure whether tests are necessarily always run with debug assertions.

Reply:
But won't the asserts affect performance and may lead to false results?

Reply:
those are setup code which are not included during benchmarking so there is no worry about measurement impact